lib/devise/multi_email.rb

```diff
@@ -1,6 +1,146 @@
 require 'devise/multi_email/version'
 require 'devise'
 
+module Devise
+  module MultiEmail
+
+    def self.configure(&block)
+      yield self
+    end
+
+    def self.parent_association_name
+      @parent_association_name ||= :user
+    end
+
+    def self.parent_association_name=(name)
+      @parent_association_name = name.try(:to_sym)
+    end
+
+    def self.emails_association_name
+      @emails_association_name ||= :emails
+    end
+
+    def self.emails_association_name=(name)
+      @emails_association_name = name.try(:to_sym)
+    end
+
+    def self.primary_email_method_name
+      @primary_email_method_name ||= :primary_email
+    end
+
+    def self.primary_email_method_name=(name)
+      @primary_email_method_name = name.try(:to_sym)
+    end
+
+    module ParentModelExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        _multi_email_emails_association_class.send :include, EmailModelExtensions
+
+        alias_method _multi_email_primary_email_method_name, :_multi_email_find_primary_email
+      end
+
+      # Gets the primary email record.
+      def _multi_email_filtered_emails
+        _multi_email_emails_association.reject(&:destroyed?).reject(&:marked_for_destruction?)
+      end
+
+      # Gets the primary email record.
+      def _multi_email_find_primary_email
+        _multi_email_filtered_emails.find(&:primary?)
+      end
+
+      def _multi_email_change_email_to(new_email)
+        valid_emails = _multi_email_filtered_emails
+        formatted_email = self.class.send(:devise_parameter_filter).filter(email: new_email)[:email]
+
+        # mark none as primary when set to nil
+        if new_email.nil?
+          valid_emails.each{|record| record.primary = false}
```

Comment:
OK, now you unmark the email as primary when the new email is nil.  I am okay with this, but think we should update the changelog.